src/modules-load/modules-load.c

```diff
@@ -57,6 +72,16 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                         return r;
         }
 
+        if (proc_cmdline_key_streq(key, "module_blacklist")) {
```

Comment:
This should be handled in `module_load_and_warn()`, as it is called also by PID1 and udevd.

Reply:
Would it be a good idea to copy over the `parse_cmdline`-related functions to `module-util.c`?

Reply:
I mean, the kernel command line option is parsed when the kmod returns -EPERM.
So, please move the parser to module-util.c, and call it when loading a module causes -EPERM.